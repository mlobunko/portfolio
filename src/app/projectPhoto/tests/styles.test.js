import React from "react";
import { shallow } from "enzyme";
import "enzyme-to-json";

import {
  StyledDescription,
  StyledNameOfProject,
  StyledProjectPhoto,
  StyledText
} from "../styles";

describe("StyledDescription", () => {
  it("renders correctly", () => {
    const wrapper = shallow(<StyledDescription />);
    expect(wrapper).toMatchSnapshot();
  });
});

describe("StyledNameOfProject", () => {
  it("renders correctly", () => {
    const wrapper = shallow(<StyledNameOfProject />);
    expect(wrapper).toMatchSnapshot();
  });
});

describe("StyledProjectPhoto", () => {
  it("renders correctly, when textColor=white, textUp=true", () => {
    const props = {
      image: "123",
      textColor: "white",
      textUp: true
    };
    const wrapper = shallow(<StyledProjectPhoto {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it("renders correctly, when textColor=false, textUp=false", () => {
    const props = {
      image: "123"
    };
    const wrapper = shallow(<StyledProjectPhoto {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});

describe("StyledText", () => {
  it("renders correctly, when textUp true", () => {
    const props = {
      textUp: true
    };
    const wrapper = shallow(<StyledText {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it("renders correctly", () => {
    const wrapper = shallow(<StyledText />);
    expect(wrapper).toMatchSnapshot();
  });
});
