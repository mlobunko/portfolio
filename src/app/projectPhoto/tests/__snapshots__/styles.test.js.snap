// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StyledDescription renders correctly 1`] = `
<div
  className="sc-bdVaJa fWVUsH"
/>
`;

exports[`StyledNameOfProject renders correctly 1`] = `
<h2
  className="sc-bwzfXH lfmvEF"
/>
`;

exports[`StyledProjectPhoto renders correctly, when textColor=false, textUp=false 1`] = `
<section
  className="sc-htpNat dqjBye"
/>
`;

exports[`StyledProjectPhoto renders correctly, when textColor=white, textUp=true 1`] = `
<section
  className="sc-htpNat efbeUJ"
/>
`;

exports[`StyledText renders correctly 1`] = `
<div
  className="sc-bxivhb hLTBkS"
/>
`;

exports[`StyledText renders correctly, when textUp true 1`] = `
<div
  className="sc-bxivhb cwszbi"
/>
`;
